sdk/eventhub/azure-eventhub/azure/eventhub/_producer_client.py

```diff
@@ -360,6 +444,40 @@ def get_partition_properties(self, partition_id):
             partition_id
         )
 
+    def get_partition_publishing_properties(self, partition_id):
+        # type: (str) -> dict
+        """Get the information about the state of publishing for a partition as observed by
+        the `EventHubProducerClient`. This data can always be read, but will only be populated with
+        information relevant to the active features for the producer client.
+
+            - `enable_idempotent_publishing` (bool)
+            - `partition_id` (str)
+            - `last_published_sequence_number` (Optional[int])
+            - `producer_group_id` (Optional[int])
+            - `owner_level` (Optional[int]
+
+        :param partition_id: The target partition ID.
+        :type partition_id: str
+        :rtype: dict
+        """
+        # pylint:disable=protected-access
+        output = {
+            'enable_idempotent_publishing': self._config.enable_idempotent_partitions,
+            'partition_id': partition_id,
+            'last_published_sequence_number': None,
```

Comment:
if the counterpart is called `starting_published_sequence_number` then should this be called `ending_published_sequence_number` for consistency? or `first_published_sequence_number` if you decide to keep `last_published_sequence_number`.
(Might not be the most important/relevant of details.)

Reply:
looks like in the sample you say `ending`

Reply:
So personally I think `last_published_sequence_number` is more suitable for a `EventHubProducerClient` and in contrast `ending_published_sequence_number` sounds reasonable to a `EventDataBatch` which has the `starting_published_sequence_number`.

But I think this a good starter for discussion on the prefix of those two object, I have created a separate issue for the naming discussion: https://github.com/Azure/azure-sdk-for-python/issues/16994, feel free to leave thoughts

Reply:
both of those sound reasonable to me